### DISCUSSION TITLE
glm-js: Sunset and Future Options

### DISCUSSION THREAD

**Discussion by humbletim** [humbletim/glm-js#19] [General]
Hi everyone,

I'm Tim, the original developer of glm-js. While I haven't actively worked on the project in over 7 years, I noticed some recent activity and wanted to provide an update.

glm-js has achieved its initial goal of bringing GLSL math operations to JavaScript. However, I no longer have the time or resources to maintain it myself.

Before archiving the project, I'd like to explore potential options for its future:

Community adoption: If there's interest, I'm open to handing the project over to a capable maintainer.
Alternatives: I'm currently considering compiling glm.cpp to Wasm for use in JavaScript. Are there other alternatives you'd recommend?
Please share your thoughts and suggestions in the comments below. I'll be monitoring this discussion for the next few weeks before deciding on the project's future.

Thank you for your interest in glm-js!

Best regards,

-Tim

**Comment by davidson16807** [comment]
I will make myself available as a comaintainer if that's desired. I've mentioned before (#17) that I have a vested interest in expanding the capabilities of this library, namely so that I can make it easier to develop another application ([here](https://github.com/davidson16807/cdcraft)) of my own. Moreover, I do believe in the value of a common, multi-language API for something so basic as 2d/3d math, and glsl is the obvious basis for it, given the ubiquity of OpenGL/WebGL and its imitations. It's best that there be one single obvious library for glsl-like behavior in each language, and for that to be the case, we need to make sure those libraries are as comprehensive, correct, unopinionated, and well maintained as possible.

Please note that I'm already spread rather thin with projects that I maintain. I'm not yet familiar with the code base. It would be nice for there to be some kind of on-boarding if possible. I also suspect it will be awhile before there is enough need from my own projects to drive development of the 2d glm functionality that I've been hoping for. That said, I also know that it can be hard to find people to act as maintainers, so if that all sounds fine and nobody else steps forward, then let me know how to proceed.
